### DISCUSSION TITLE
File 'le_max_key_size' in directory 'hci0' already present!

### DISCUSSION THREAD

**Discussion by balonchiks** [mKeRix/room-assistant#377] [Q&A] [answered]
Hi, i am using a cluster of 5 raspberry pi's (rpi4, rpi3 and 3 x rpi zero w) and all of them have dmesg flooded with this message:

`
debugfs: File 'le_min_key_size' in directory 'hci0' already present!`
`
debugfs: File 'le_max_key_size' in directory 'hci0' already present!`

the detection seems to work fine and is not affected by this message, but still interesting to see why is this and is there a fix for this?

it seems to happen every time when a bluetooth scan is issued by the pi.

thanks

**Comment by jdiegmueller** [comment]
Are you using the built-in Bluetooth adapters, or external?  

I've been kicking the tires on room-assistant for a few days.  At first I assumed it would be wise to use external (USB) Bluetooth adapters, and I kept getting the "le_min_key_size" stuff.  Unlike your scenario where it seems to be cosmetic, in my scenario (BLE + Radbeacon Dots) the le_xxx_key_size messages effectively meant BLE tag detection would halt.  

I'm curious if this issue is specific to external Bluetooth adapters or if folks are seeing it on the built-in adapters as well.

**Reply by balonchiks** [reply]
> Are you using the built-in Bluetooth adapters, or external?

built-in, just the raspberry pi zero w and clean raspbian os install.

**Comment by fidonet** [comment]
I have a cluster of Zero W's and a Pi4 ... I get this on all of them, and they're all using the built in BT/Wifi adapter

[42303.295408] Bluetooth: hci0: unexpected event for opcode 0x200b
[42303.417150] debugfs: File 'le_min_key_size' in directory 'hci0' already present!
[42303.417169] debugfs: File 'le_max_key_size' in directory 'hci0' already present!

[81222.055793] debugfs: File 'le_min_key_size' in directory 'hci0' already present!
[81222.055819] debugfs: File 'le_max_key_size' in directory 'hci0' already present!
[83850.550618] debugfs: File 'le_min_key_size' in directory 'hci0' already present!
[83850.550643] debugfs: File 'le_max_key_size' in directory 'hci0' already present!

etc

**Comment by mKeRix** [comment, marked as answer]
Unfortunately I don't think there is a fix for this yet, at least none that I know of. I see the same errors across my Pis too. It should not affect the functionality of room-assistant though, as you mentioned.

There is a [reported bug](https://bugzilla.kernel.org/show_bug.cgi?id=204765) with the same error message that was supposedly fixed in v5.6 of the Linux kernel, but I'm not sure if that is the same issue or just the same message with different causes. The Pi that I just re-installed two days ago is currently sitting at kernel v5.4, so it seems that v5.6 isn't available as a regular software update yet.
